File: 0682/Sesión 04 - lib java/Publicaciones_v03/src/main/java/pe/edu/pucp/inf30/sesiones/publicaciones03/publicaciones/estandardAPA/Articulo.java
package pe.edu.pucp.inf30.sesiones.publicaciones03.publicaciones.estandardAPA;

import pe.edu.pucp.inf30.sesiones.publicaciones03.publicaciones.Publicacion;

public class Articulo extends Publicacion {

    private String revista;
    private Integer volumen;
    private Integer numero;
    private String mes;

    public Articulo(String nombre, String nombre_completo_autor, Integer año, String revista, Integer volumen, Integer numero, String mes) {
        super(nombre, nombre_completo_autor, año);
        this.revista = revista;
        this.volumen = volumen;
        this.numero = numero;
        this.mes = mes;
    }

    public Articulo(String nombre, String nombres_completos_autores[], Integer año, String revista, Integer volumen, Integer numero, String mes) {
        super(nombre, nombres_completos_autores, año);
        this.revista = revista;
        this.volumen = volumen;
        this.numero = numero;
        this.mes = mes;
    }

    @Override
    public String toString() {
        String referencia = "[" + Publicacion.orden_presentacion.toString() + "]";
        for (int i = 0; i < cantidad_autores; i++) {
            if (i != 0) {
                if (i == cantidad_autores - 1) {
                    referencia = referencia.concat(" and ");
                } else {
                    referencia = referencia.concat(", ");
                }
            }
            referencia = referencia.concat(autores.get(i).toString());
        }
        referencia = referencia.concat(", \"");
        referencia = referencia.concat(nombre);
        referencia = referencia.concat("\", "); 
        referencia = referencia.concat(revista);
        referencia = referencia.concat(", vol. ");
        referencia = referencia.concat(volumen.toString());
        referencia = referencia.concat(", no. ");
        referencia = referencia.concat(numero.toString());
        referencia = referencia.concat(", ");
        referencia = referencia.concat(mes);
        referencia = referencia.concat(" ");
        referencia = referencia.concat(año.toString());
        referencia = referencia.concat(". ");
        return referencia;
    }public void referenciar() {
        String referencia = "";
        for (int i = 0; i < cantidad_autores; i++) {
            if (i != 0) {
                if (i == cantidad_autores - 1) {
                    referencia = referencia.concat(" and ");
                } else {
                    referencia = referencia.concat(", ");
                }
            }
            referencia = referencia.concat(autores.get(i).toString());
        }
        referencia = referencia.concat(". (");
        referencia = referencia.concat(año.toString());
        referencia = referencia.concat("). ");
        referencia = referencia.concat(nombre);
        referencia = referencia.concat(". ");
        referencia = referencia.concat(revista);
        referencia = referencia.concat(", ");
        referencia = referencia.concat(volumen.toString());
        referencia = referencia.concat("(");
        referencia = referencia.concat(numero.toString());
        referencia = referencia.concat(").");
        System.out.println(referencia);
    }
}

File: 0682/Sesión 05 - lib C#/Publicaciones_v03/Publicaciones_v03/Publicaciones/EstandardAPA/Articulo.cs
using Publicaciones;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Publicaciones.EstandardAPA
{
    internal class Articulo: Publicacion
    {
        private string revista;
        private int volumen;
        private int numero;
        private string mes;

        public Articulo(string nombre, string nombre_completo_autor, int año, string revista, int volumen, int numero, string mes)
            : base(nombre, nombre_completo_autor, año)
        {
            this.revista = revista; 
            this.volumen = volumen;
            this.numero = numero;
            this.mes = mes;
        }

        public Articulo(string nombre, string[] nombres_completos_autores, int año, string revista, int volumen, int numero, string mes)
           : base(nombre, nombres_completos_autores, año)
        {
            this.revista = revista;
            this.volumen = volumen;
            this.numero = numero;
            this.mes = mes;
        }

        public override string ToString()
        {
            string referencia = "";
            for (int i = 0; i < this.autores.Count; i++)
            {
                if (i != 0)
                {
                    if (i == this.cantidad_autores - 1)
                        referencia += " and ";
                    else
                        referencia += ", ";
                }
                referencia += this.autores[i].ToString();
            }
            referencia += ". (";
            referencia += this.año;
            referencia += "). ";
            referencia += nombre;
            referencia += ". ";
            referencia += revista;
            referencia += ", ";
            referencia += this.volumen;
            referencia += " (";
            referencia += this.numero;
            referencia += ").";
            return referencia;
        }
    }
}
